(doc) remove reference to pre-assoctypes type param `R`

Ryman opened the pull request.

Ryman committed:
(doc) remove reference to pre-assoctypes type param `R`

reem commented on src/lib.rs:
```diff
@@ -36,15 +36,13 @@ pub trait Extensible {
     /// If plugin creation fails, an error is returned.
     ///
-    /// `P` is the plugin type.
+    /// `P` is the plugin type, `P::Value` must implement `Clone`.
```
Do you think it's valuable to say `P::Value` must be `Clone` since it's already in the type? I generally hesitate to repeat that kind of thing but if it's non-obvious we can include it.

Ryman replied:
Not really, I generally find rust docs pretty good at exposing the bound requirements, I just moved the comment that was already kind of there when talking about `R` (and it still seemed relevant).

I'm happy to remove it. Would you like me to also add a more modern `travis.yml` while I'm at it? Seems like rustup is now an interactive script.

Ryman committed:
(chore) add modern travis config

Ryman committed:
(doc) update README

Ryman commented:
@reem Forgot to ping when i pushed updates yesterday. Added a new travis setup, updated readme and then removed the comment about Clone from original. Can squash to single commit if you need.

reem commented on .travis.yml:
```diff
@@ -1,11 +1,22 @@
-os:
-  - linux
+  - |
+      cargo build &&
```
```
  - cargo build
  - cargo test
  - cargo doc
```

would be the best way to format this, I think.

reem commented on .travis.yml:
```diff
@@ -1,11 +1,22 @@
+matrix:
+  fast_finish: true
+  allow_failures:
+    - rust: nightly
```
oh, why?

Ryman replied:
Allows contributors without nightly to not worry about installing one if things fail, on the basis that anything relevant would be caught when it hits beta. I'll just remove it for now.

reem commented:
Left some comments on travis setup, everything else looks good.

Ryman commented:
Updated!

reem commented:
Great, thanks!
